js/webui.js

```diff
@@ -615,7 +615,7 @@ var theWebUI =
 
 	initSettings: function(newSettings)
 	{
-		// If when receive an empty string for the webui settings,
+		// If when receive an empty JSON object for the webui settings,
 		// We must initialize "webui.lang", so it can be set
 		if (newSettings=="{}")
```

Comment:
1) You can't compare Object with a String. I.e. you can, but you will always have 'false' here.
Try to write something like ```({}=="{}");``` in the browser console...
2) Why we need this?

Reply:
1. It works perfectly fine, I tested it. It's not an object, it's a string being sent here.

2. We can't reference "webui.lang" as a key then set it's value for a JavaScript array, if the variable is in string format.

Reply:
Yes, it does return a JavaScript object. The object may have a primitive value of `"{}"` (which is basically a string), if for some reason `gettsettings.php->WebUISettings.php` can't load the .dat file. Like for instance on the first run, when it's not preset. This is why we can compare it as a string. If the JavaScript object had a value of say `{"webui.lang": ""}` this would always return false. The JavaScript object being returned can be a simple (primitive) object or a complex object.

Reply:
> The object may have a primitive value of "{}" (which is basically a string)

Object is a object. String is a string. Object can't "have a primitive value of '{}'". Object may be an empty.
But you can't check this with a such strange way.

> Even if you're right and this is not a string, == is being used here which does not check the datatype.

This haven't relation to the question because Object cannot be brought to string by automatic way.

Yet once, open browser console and print in it 
```
({}=="{}");
```
This is a comparison of empty object and string "{}".
What do you see as result?
I see 'false'. 

If you want to check empty object you should write something like 
```
(JSON.stringify({})=="{}");
```
or [use jQuery.isEmptyObject()](https://api.jquery.com/jquery.isemptyobject/)

Reply:
Alright, I used `JSON.stringify`. I don't think this is required, but it doesn't matter because it works ether way.